File: mgl-webapp/src/main/webapp/resources/static/js/game.controller.js
'use strict';

angular.module('MGLApp').controller('GameController',
		[ 'GameService', function(GameService) {
			var self = this;
			self.game = {
				id : '',
				name : '',
				genre : ''
			};
			self.games = [];

			self.fetchAllGames = function(){
				GameService.fetchAllGames().then(function(data) {
					self.games = data;
				}).finally( function() {
					self.game= {};
				});
			}

			self.addGame = function(){
				return GameService.createGame(self.game).then( function() {
					self.fetchAllGames();
				});
			}
			
			self.loadGame = function(newGame) {
				self.game = angular.copy(newGame);
			}
			
			self.updateGame = function(){
				return GameService.updateGame(self.game).then( function() {
					self.fetchAllGames();
				});
			}
			
			self.deleteGame = function(game) {
				return GameService.deleteGame(game.id).then( function() {
					self.fetchAllGames();
				});
			}
			

			self.fetchAllGames();
		} ]);

File: mgl-webapp/src/main/java/com/organization/mvcproject/controller/GameController.java
package com.organization.mvcproject.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.organization.mvcproject.api.service.GameService;
import com.organization.mvcproject.model.GameImp;

@RestController
@RequestMapping(value = "/game")
public class GameController {

	@Autowired
	private GameService gameService;
	
	@GetMapping(value = "/")
	public ResponseEntity<List<GameImp>> fetchAllGames() {
		return new ResponseEntity<List<GameImp>>(gameService.retrieveAllGames(), HttpStatus.OK);
	}

	@PostMapping(value = "/", consumes = MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity<Void> createGame(@RequestBody GameImp game) {
		gameService.saveGame(game);
		return new ResponseEntity<Void>(HttpStatus.CREATED);
	}
	
	@PutMapping(value = "/", consumes = MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity<?> updateGame(@RequestBody GameImp game) {
		GameImp savedGame = gameService.saveGame(game);
		return new ResponseEntity<>(savedGame, HttpStatus.OK);
	}
	
	@DeleteMapping(value = "/{id}")
	public ResponseEntity<?> deleteGame(@PathVariable(value="id") Long id){
		return new ResponseEntity<>(gameService.deleteGame(id), HttpStatus.OK);
	}
}